### ISSUE TITLE
Using "find" when a directory is in the root of an archive doesn't work consistently

### ISSUE THREAD

**Issue by plague006** [KSP-CKAN/CKAN#1477]
I'm not sure what the conditions of it working or failing are, but here are 2 samples of metadata. The first results in failure (no files to install) and the second in success. This issue occurs independently with netkan.exe using an otherwise valid .netkan, and with ckan.exe using an otherwise valid .ckan.

```
{
    "spec_version": "v1.4",
    "license": "CC-BY-NC-SA-4.0",
    "identifier": "ShuttleLiftingBodyCormorantAeronology",
    "$kref": "#/ckan/kerbalstuff/1134",
    "install": [
        {
            "find"       : "Cormorant Aeronology",
            "install_to" : "GameData"
        }
    ]
}
```

---

```
{
    "spec_version": "v1.4",
    "license": "CC-BY-NC-SA-4.0",
    "identifier": "CarcharothAeronautics",
    "$kref": "#/ckan/kerbalstuff/1127",
    "install": [
        {
            "find"       : "Carcharoth Aeronautics",
            "install_to" : "GameData"
        }
    ]
}
```

**Comment by pjf** [comment]
I have a hunch that some zip-files explicitly include an entry for the root path, and some don't, and we're failing in the second case. `ModuleInstallDescriptor.ConvertFindToFile()` is quite possibly where this bug is lurking.

**Comment by pjf** [comment]
In the case of KSP-CKAN/NetKAN#4169 I'm not really sure what's going on. The archive starts with `GameData/StarWarsMod` (for 1.1.2 which I downloaded), which I would expect a `find` to work on just fine... Seeing if I can reproduce the original bug now.

**Comment by pjf** [comment]
I don't have a fix yet (patches welcome!) but I can tell you exactly what's happening:

The BD StarWars Mod looks like this:

```
  Length      Date    Time    Name
---------  ---------- -----   ----
     6785  2016-03-01 18:34   StarWarsMod/Flags/KS Logo.png
    21302  2016-05-30 11:46   StarWarsMod/Flags/Parts/AmmoBox/model.mu
    17003  2016-05-30 11:46   StarWarsMod/Flags/Parts/AmmoBox/texture.png
     1016  2016-05-30 11:46   StarWarsMod/Flags/Parts/AmmoBox/TibannaGas.cfg
     1855  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannonb.cfg
   131272  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/model.mu
    67963  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/muzzle.png
     1595  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/smoke.png
    29047  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/tex_browninganm2.png
    70656  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/Sounds/fire.wav
     1779  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/lasercannong.cfg
   131272  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/model.mu
    18482  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/muzzle.png
     1595  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/smoke.png
    29101  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/tex_browninganm2.png
    55340  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/Sounds/fire.wav
     1923  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/lasercannongeneralgrievous/lasercannongeneralgrievous.cfg
   131272  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/lasercannongeneralgrievous/model.mu
     5940  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/lasercannongeneralgrievous/muzzle.png
     1595  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/lasercannongeneralgrievous/smoke.png
    29049  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/lasercannongeneralgrievous/tex_browninganm2.png
    48428  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/lasercannongeneralgrievous/Sounds/fire.wav
     1826  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/lasercannongeneralgrievous/lasercannonr/lasercannonr.cfg
   131272  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/lasercannongeneralgrievous/lasercannonr/model.mu
    12445  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/lasercannongeneralgrievous/lasercannonr/muzzle.png
     1595  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/lasercannongeneralgrievous/lasercannonr/smoke.png
    29082  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/lasercannongeneralgrievous/lasercannonr/tex_browninganm2.png
    72728  2016-05-30 11:47   StarWarsMod/Flags/Parts/AmmoBox/lasercannonb/lasercannong/lasercannongeneralgrievous/lasercannonr/Sounds/fire.wav
      127  2016-05-30 11:46   StarWarsMod/Flags/Parts/Resources/Ammo.cfg
---------                     -------
  1053345                     29 files
```

In `ConvertFindToFile` we:

1. Take a list of all zip entries
1. Snip off the file when applicable ( `StarWarsMod/Flags/Parts/Resources/Ammo.cfg` -> `StarWarsMod/Flags/Parts/Resources`)
1. Find the shortest entry that ends with the directory we're after.

The gotcha is that there are no files in the `StarWarsMod` directory itself, there are only other directories, and so if you look at the above list you'll see we never end up with a path ending with the directory we're looking for, and hence the find process fails.

The solution would be to add a step between 2 and 3 which also expands out all the parent directories.

Should be a Simple Matter Of Code, right? :)

**Comment by HebaruSan** [comment]
Fixed by #2120 (would have linked it at the time if I had known of this issue report).
